apps/common-app/src/examples/RuntimeTests/tests/animations/withDelay/addDelays.test.tsx

```diff
@@ -0,0 +1,147 @@
+    const widthActiveSV = useSharedValue(0);
+    const widthPassiveSV = useSharedValue(0);
+
+    const refActive = useTestRef(WIDTH_COMPONENT_ACTIVE_REF);
+    const refPassive = useTestRef(WIDTH_COMPONENT_PASSIVE_REF);
+    const refInline = useTestRef(WIDTH_COMPONENT_INLINE_REF);
+
+    const styleActive = useAnimatedStyle(() => {
+      return {
+        width: widthPassiveSV.value,
+      };
+    });
+    const stylePassive = useAnimatedStyle(() => {
+      if (delays.length === 1) {
+        return {
+          width: withDelay(delays[0], withTiming(widthActiveSV.value, { duration: 100 })),
+        };
+      }
+      if (delays.length === 3) {
+        return {
+          width: withDelay(
+            delays[0],
+            withDelay(delays[1], withDelay(delays[2], withTiming(widthActiveSV.value, { duration: 100 }))),
+          ),
+        };
+      }
+      return {};
+    });
+
+    useEffect(() => {
+      widthActiveSV.value = 300;
+    }, [widthActiveSV]);
+
+    useEffect(() => {
+      if (delays.length === 1) {
+        widthPassiveSV.value = withDelay(delays[0], withTiming(300, { duration: 100 }));
+      }
+      if (delays.length === 3) {
+        widthPassiveSV.value = withDelay(
+          delays[0],
+          withDelay(delays[1], withDelay(delays[2], withTiming(300, { duration: 100 }))),
+        );
+      }
+    }, [widthPassiveSV, delays]);
+
+    return (
+      <View style={styles.container}>
+        <Animated.View ref={refActive} style={[styles.animatedBox, { backgroundColor: 'powderblue' }, styleActive]} />
+        <Animated.View
+          ref={refPassive}
+          style={[styles.animatedBox, { backgroundColor: 'cornflowerblue' }, stylePassive]}
+        />
+        <Animated.View
+          ref={refInline}
+          style={[styles.animatedBox, { backgroundColor: 'steelblue', width: widthPassiveSV }]}
+        />
+      </View>
+    );
+  };
+
+  async function getSnapshotUpdates(delays: [number] | [number, number, number]) {
+    await mockAnimationTimer();
+    const updatesContainer = await recordAnimationUpdates();
+    await render(<WidthComponent delays={delays} />);
+    const delaySum = delays.reduce((current, sum) => sum + current, 0);
+    await wait(150 + delaySum);
+
+    const updates = COMPONENT_REFS.map(ref => updatesContainer.getUpdates(getTestComponent(ref)));
+    const updatesNative = [];
+    for (const ref of COMPONENT_REFS) {
+      const newUpdates = await updatesContainer.getNativeSnapshots(getTestComponent(ref));
+      updatesNative.push(newUpdates);
+    }
+    await unmockAnimationTimer();
+    await clearRenderOutput();
+
+    return [updates, updatesNative];
+  }
+
+  test.each([
+    [10, 20, 30],
+    [10, 0, 30],
+    [0, 30, 30],
+    [40, 30, 0],
+
+    // [0, 0, 30], This tests don't pass (probably we have some unexpected delay)
+    // [40, 0, 0],
+    // [0, 55, 0],
+    // [0, 0, 0],
```

Comment:
What is happening in this case?

Reply:
There is a translation bigger than one frame (two frames)